test/runner/run.js

```diff
@@ -4,38 +4,40 @@ import { getLatestBrowser } from "./browserstack/api.js";
 import { buildBrowserFromString } from "./browserstack/buildBrowserFromString.js";
 import { localTunnel } from "./browserstack/local.js";
 import { reportEnd, reportTest } from "./reporter.js";
-import {
-	cleanupAllWorkers,
-	cleanupWorker,
-	debugWorker,
-	retryTest,
-	touchWorker
-} from "./browserstack/workers.js";
 import { createTestServer } from "./createTestServer.js";
 import { buildTestUrl } from "./lib/buildTestUrl.js";
 import { generateHash, printModuleHashes } from "./lib/generateHash.js";
 import { getBrowserString } from "./lib/getBrowserString.js";
-import { addRun, runFullQueue } from "./queue.js";
 import { cleanupAllJSDOM, cleanupJSDOM } from "./jsdom.js";
 import { modules as allModules } from "./modules.js";
+import { cleanupAllBrowsers, touchBrowser } from "./browserstack/browsers.js";
+import {
+	addBrowserStackRun,
+	getNextBrowserTest,
+	retryTest,
+	runAllBrowserStack
+} from "./browserstack/queue.js";
+import { addSeleniumRun, runAllSelenium } from "./selenium/queue.js";
 
 const EXIT_HOOK_WAIT_TIMEOUT = 60 * 1000;
 
 /**
  * Run modules in parallel in different browser instances.
  */
-export async function run( {
-	amd,
-	browsers: browserNames,
-	browserstack,
-	concurrency,
-	debug,
-	headless,
-	isolate = true,
-	modules = [],
-	retries = 0,
-	verbose
-} = {} ) {
+export async function run( options = {} ) {
+	let {
+		amd,
+		browsers: browserNames,
+		browserstack,
+		debug,
+		headless,
+		isolate,
+		modules = [],
+		retries = 0,
+		runId,
+		verbose
+	} = options;
```

Comment:
This method is risky IMO. The defaults will only apply to destructured variables like `modules` or `retries` but the `options` object contains them as they were passed. I suspect this may be a problem since you pass `options` to a few APIs as-is.

It'd be safer to keep destructuring directly in parameters. If you then need the `options` object to be fully passed in some places, you can always define it here instead of destructuring:
```js
	const options = {
		amd,
		browsers: browserNames,
		/* ... */
	};
```
The main drawback is each parameter appears twice but the risk above disappears.

Reply:
Ok, why don't I just go back to what it was before. I'll pass all params to the run functions manually again.